qrexec/policy/admin.py

```diff
@@ -87,7 +87,13 @@ def handle_request(
         Throws PolicyAdminException in case of user error.
         """
 
-        assert all(char in RPCNAME_ALLOWED_CHARSET for char in arg)
+        try:
+            assert all(char in RPCNAME_ALLOWED_CHARSET for char in arg)
+        except AssertionError as ex:
```

Comment:
If changing to a different exception, I'd prefer simply `if not all(....): raise` instead of converting one exception to another.

Anyway, as @ben-grande noticed, this isn't exactly correct. This verifies call argument, but depending on specific call, it may be a file name (most of the methods here), or a service name (at least `policy.GetFiles`). The latter one already verifies its argument separately, maybe better to include also file name verification (with appropriate error message) in the `_get_path` function with rules guarding file names?

Reply:
> maybe better to include also file name verification (with appropriate error message) in the `_get_path` function with rules guarding file names?

There is a slight problem with this approach. `handle_request` and its validation happens before `_get_path` function call. It may be better to implement both type of validation within `handle_request` itself. I will submit the amended PR shortly.

Reply:
@marmarek 
I believe this is ready (including updated unittests). The failed test appears to be due to a temporal network connection. I do not have access to RetryPipelineFailed command.

Reply:
> `handle_request` and its validation happens before `_get_path` function call.

Technically, it's only partially true. The final part of `handle_request` calls the actual handler that internally calls `_get_path`. So, as long as validation in `handle_request` itself is not stricter than one needed in `_get_path`, it should be okay.